packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,216 @@
+import { GuardManager, Guard as IGuard } from "safe-contracts/base/GuardManager.sol";
+
+// Interfaces
+import { ISemver } from "interfaces/universal/ISemver.sol";
+
+/// @title TimelockGuard
+/// @notice This guard provides timelock functionality for Safe transactions
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this guard using GuardManager.setGuard()
+///      2. The Safe must then configure the guard by calling configureTimelockGuard()
+contract TimelockGuard is IGuard, ISemver {
+    /// @notice Configuration for a Safe's timelock guard
+    struct GuardConfig {
+        uint256 timelockDelay;
+    }
+
+    /// @notice Mapping from Safe address to its guard configuration
+    mapping(address => GuardConfig) public safeConfigs;
+
+    /// @notice Mapping from Safe address to its current cancellation threshold
+    mapping(address => uint256) public safeCancellationThreshold;
+
+    /// @notice Error for when guard is not enabled for the Safe
+    error TimelockGuard_GuardNotEnabled();
+
+    /// @notice Error for when Safe is not configured for this guard
+    error TimelockGuard_GuardNotConfigured();
+
+    /// @notice Error for when attempt to clear guard while it is still enabled for the Safe
+    error TimelockGuard_GuardStillEnabled();
+
+    /// @notice Error for invalid timelock delay
+    error TimelockGuard_InvalidTimelockDelay();
+
+    /// @notice Emitted when a Safe configures the guard
+    event GuardConfigured(address indexed safe, uint256 timelockDelay);
+
+    /// @notice Emitted when a Safe clears the guard configuration
+    event GuardCleared(address indexed safe);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice Returns the timelock delay for a given Safe
+    /// @dev MUST never revert
+    /// @param _safe The Safe address to query
+    /// @return The timelock delay in seconds
+    function viewTimelockGuardConfiguration(address _safe) public view returns (uint256) {
+        // Q: What should this return if the guard is not enabled?
+        return safeConfigs[_safe].timelockDelay;
+    }
+
+    /// @notice Configure the contract as a timelock guard by setting the timelock delay
+    /// @dev MUST allow an arbitrary number of Safe contracts to use the contract as a guard
+    /// @dev MUST revert if the contract is not enabled as a guard for the Safe
+    /// @dev MUST revert if timelock_delay is longer than 1 year
+    /// @dev MUST set the caller as a Safe
+    /// @dev MUST take timelock_delay as a parameter and store it as related to the Safe
+    /// @dev MUST emit a GuardConfigured event with at least timelock_delay as a parameter
+    /// @param _timelockDelay The timelock delay in seconds
+    function configureTimelockGuard(uint256 _timelockDelay) external {
+        // Validate timelock delay - must be non-zero and not longer than 1 year
```

Comment:
I'm not sure that we want to prevent a timelock being configured with zero delay.
- I don't see the possible harm.
- If we want a zero delay, it might be better for integration to have just that than not enabling the guard, so that it is still schedule -> exec

Not too bothered either way, though. But if we decide to not allow zero delays I'll add it to the spec (maybe with a reason?)